### ISSUE TITLE
chapter 4 - code incomplete

### ISSUE THREAD

**Issue by fileung** [PacktPublishing/Hands-On-Generative-Adversarial-Networks-with-Keras#5]
Hi, 
In the code file chapter4.py
without any change/fix, the code will return an IndentationError, at line:
q_y = np.mean(p_y_given_x, axis=0)

after fixing the indentation, there are more errors.
It looks like the code is incomplete, mainly all the imports are missing.

**Comment by rafaelvalle** [comment]
@fileung, these excerpts should be added to your existing training code, for example, the data and the models are available.
For example, [this line](https://github.com/PacktPublishing/Hands-On-Generative-Adversarial-Networks-with-Keras/blob/master/Chapter04/chapter4.py#L2) refers to the training data and [this line](https://github.com/PacktPublishing/Hands-On-Generative-Adversarial-Networks-with-Keras/blob/master/Chapter04/chapter4.py#L4) refers to a Generator.

**Comment by fileung** [comment]
@rafaelvalle 
Is it possible to provide the complete code for chapter 4? it will be much easier when the reader just trying to execute the code provided and see some results, without changing the default code (i did and got code exceptions). thanks.

**Comment by rafaelvalle** [comment]
Our goal is to have readers learn how to incorporate bits of code into their own stack. By providing a complete solution, the readers would not learn this valuable skill. 

Having said that, please let us know if you have any doubts! I'm sure we can help you learn how to use those bits of code and ideas in your own projects.

**Comment by rafaelvalle** [comment]
Closing due to inactivity.
